packages/trace-viewer/src/ui/uiModeTraceView.tsx

```diff
@@ -108,11 +108,24 @@ const outputDirForTestCase = (testCase: reporterTypes.TestCase): string | undefi
   return undefined;
 };
 
-async function loadSingleTraceFile(url: string): Promise<MultiTraceModel> {
+async function loadSingleTraceFile(traceURL: URL): Promise<MultiTraceModel> {
   const params = new URLSearchParams();
-  params.set('trace', url);
+  params.set('trace', formatUrl(traceURL).toString());
   params.set('limit', '1');
   const response = await fetch(`contexts?${params.toString()}`);
   const contextEntries = await response.json() as ContextEntry[];
   return new MultiTraceModel(contextEntries);
 }
+
+function formatUrl(traceURL: URL) {
+  if (traceURL.hostname === 'dropbox.com')
+    traceURL.hostname = 'dl.dropboxusercontent.com';
+
+  return traceURL;
+}
+
+function filePathToTraceURL(path: string) {
+  const url = new URL('/trace/file', location.href);
```

Comment:
I wonder whether this should be `/trace/file` or just `file`, like in line 115? For example, when running ui mode with `--ui-host` option through codespaces, I am not sure whether UI mode frontend is always served as `<host>:<port>/trace...` without any path prefix.

Reply:
that's a good point. I've updated this in https://github.com/microsoft/playwright/pull/33344/commits/69ba2d1d66ccf30239973e0f95911658974733df.